uportal-war/src/main/java/org/jasig/portal/portlets/search/SearchPortletController.java

```diff
@@ -719,12 +723,18 @@ private void addSearchResults(SearchResults portletSearchResults, PortalSearchRe
      */
     protected void modifySearchResultLinkTitle(SearchResult result, final HttpServletRequest httpServletRequest,
                                                final IPortletWindowId portletWindowId) {
-        if (result.getType().size() > 0 && searchResultsLinkTextPortletTitleSubstitutionSet.contains(result.getType().get(0))) {
-            final IPortletWindow portletWindow = this.portletWindowRegistry.getPortletWindow(httpServletRequest, portletWindowId);
+        // If the title contains a SpEL expression, parse it with the portlet definition in the evaluation context.
+        if (result.getType().size() > 0 && result.getTitle().contains("${")) {
+            final IPortletWindow portletWindow = this.portletWindowRegistry.getPortletWindow(httpServletRequest,
+                    portletWindowId);
             final IPortletEntity portletEntity = portletWindow.getPortletEntity();
             final IPortletDefinition portletDefinition = portletEntity.getPortletDefinition();
-            String portletTitle = portletDefinition.getTitle();
-            result.setTitle(portletTitle);
+            final SpELEnvironmentRoot spelEnvironment = new SpELEnvironmentRoot(portletDefinition);
+            try {
+                result.setTitle(spELService.getValue(result.getTitle(), spelEnvironment));
+            } catch (SpelParseException e) {
+                result.setTitle("Invalid Spring EL expression '" + result.getTitle()+"'");
```

Comment:
Hmm.  Should the failure case be to show this message to users as the title of a search result?  Seems like maybe the failure should be logged and the search result should fall back on some kind of best-under-the-circumstances title ("(Unknown content title)")?